File: ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/EncumbranceGroup.java
package net.corda.v5.ledger.utxo;

import net.corda.v5.base.annotations.CordaSerializable;
import net.corda.v5.base.annotations.DoNotImplement;
import org.jetbrains.annotations.NotNull;

/**
 * Identifies the encumbrance of a [TransactionState]
 * <p>
 * The encumbrance is identified by a tag. The encumbrance group has the tag and the size of the encumbrance group,
 * i.e. the number of states encumbered with the same tag in the same transaction. This allows to easily check
 * that all states of one encumbrance group are present.
 */
@DoNotImplement
@CordaSerializable
public interface EncumbranceGroup {

    /**
     * Gets the encumbrance tag.
     *
     * @return Returns the encumbrance tag.
     */
    @NotNull
    String getTag();

    /**
     * Gets the number of states encumbered with the tag of this group.
     *
     * @return Returns the number of states encumbered with the tag of this group.
     */
    int getSize();
}

File: ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/EncumbranceGroup.kt
package net.corda.v5.ledger.utxo

import net.corda.v5.base.annotations.CordaSerializable
import net.corda.v5.base.annotations.DoNotImplement

/**
 * Identifies the encumbrance of a [TransactionState]
 *
 * The encumbrance is identified by a tag. The encumbrance group has the tag and the size of the encumbrance group,
 * i.e. the number of states encumbered with the same tag in the same transaction. This allows to easily check
 * that all states of one encumbrance group are present.
 *
 * @property tag The encumbrance tag
 * @property size The number of states encumbered with the tag of this group
 */
@DoNotImplement
@CordaSerializable
interface EncumbranceGroup {
    val tag: String
    val size: Int
}